### PR TITLE
Codechange: math functions - use cpp-style casts

### PR DESCRIPTION
**Pull request by nikolas**

### CODE REVIEW

**Comment by glx22** [comment]
Of course commit checker fails because `#` should move to first column when the line is modified.
`<tabs>#xxx` must now be `#<tabs>xxx`

**Comment by nikolas** [comment]
Ah i see, i just found an old conversation about this here: https://github.com/OpenTTD/OpenTTD/pull/7407#issuecomment-475974158

I'll make that change.

**Comment by glx22** [comment]
Please restore the indentation between `#` and `define`

**Comment by LordAro** [comment]
Not that I necessarily disagree with the changes, but I'd like to see some justification for them. What problem do they solve? How do they improve things?

**Comment by nikolas** [comment]
Using C++ style casts makes the intention of the code more explicit:
> The problem with C casts is the ambiguity of the operation; sometimes you are doing a conversion (e.g., (int)3.5) and sometimes you are doing a cast (e.g., (int)"hello"). Brace initialization and C++ casts can often help avoid this ambiguity. Additionally, C++ casts are more visible when searching for them.

From: https://google.github.io/styleguide/cppguide.html#Casting

That said, maybe this should be an all-or-nothing style decision. C++ casts could be used everywhere if that's decided to be an improvement, otherwise just keep everything as it is.

File: src/core/bitmath_func.hpp [hunk 1]
```diff
@@ -383,7 +383,7 @@ static inline T ROR(const T x, const uint8 n)
 	{
 #if !defined(__ICC) && defined(__GNUC__) && ((__GNUC__ > 4) || ((__GNUC__ == 4)  && __GNUC_MINOR__ >= 3))
 		/* GCC >= 4.3 provides a builtin, resulting in faster code */
-		return (uint32)__builtin_bswap32((int32)x);
+		return static_cast<uint32>(__builtin_bswap32(static_cast<int32>(x)));
```
Comment (LordAro) [inline src/core/bitmath_func.hpp:386]: According to https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html, the cast to int32 is unnecessary (or possibly wrong)

**Review by LordAro — APPROVED** [review]
Close enough

**Pull request merged**